homeassistant/components/tplink/switch.py

```diff
@@ -57,8 +61,9 @@ def add_entity(device: SmartPlug, async_add_entities):
 class SmartPlugSwitch(SwitchDevice):
     """Representation of a TPLink Smart Plug switch."""
 
-    def __init__(self, smartplug: SmartPlug):
+    def __init__(self, hass: HomeAssistantType, smartplug: SmartPlug):
```

Comment:
Don't pass in `hass`. It will be set on the entity when it has been added to home assistant.

Reply:
No longer passing hass.